plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1827,6 +1821,65 @@ public Optional<ConnectorOutputMetadata> finishInsert(
         return Optional.empty();
     }
 
+    private long getInsertCommitVersion(ConnectorSession session, DeltaLakeInsertTableHandle handle, List<DataFileInfo> dataFileInfos)
```

Comment:
In the event you retry, right now you're redoing all the work again. This is probably fine, but you could just read the new entries that you didn't validate the first time.

Reply:
Good point Alex.
Introduced `GetInsertCommitVersionExecutionContext` context holder.